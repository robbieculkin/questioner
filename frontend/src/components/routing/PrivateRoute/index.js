import React from 'react';
import { Route, Redirect } from 'react-router-dom';

// Protect Route from being explicitly accessed without
// providing the necessary state through an internal <Link />
const PrivateRoute = ({ component: Component, ...rest }) =>
  <Route {...rest} render={
    props => (
      props.location.state
        ? <Component {...props} />
        : <Redirect to='/' />
    )}
  />

export default PrivateRoute;
