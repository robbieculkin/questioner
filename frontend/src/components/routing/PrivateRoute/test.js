import React from 'react';
import { shallow } from 'enzyme';
import PrivateRoute from '.';

describe('PrivateRoute', () => {
  it('should render correctly without props', () => {
    const component = shallow(<PrivateRoute />);
    expect(component).toMatchSnapshot();
  });
});
