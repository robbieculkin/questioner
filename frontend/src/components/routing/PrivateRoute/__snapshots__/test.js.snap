// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivateRoute should render correctly without props 1`] = `
<Route
  render={[Function]}
/>
`;
